src/main/java/uk/gov/hmcts/reform/hmc/service/LinkedHearingGroupServiceImpl.java

```diff
@@ -109,21 +106,12 @@ private int getIdOccurrences(List<LinkHearingDetails> hearingDetails, String val
     }
 
     private LocalDate filterHearingResponses(HearingEntity hearingEntity) {
-        String version = hearingEntity.getLatestRequestVersion().toString();
-        Optional<HearingResponseEntity> hearingResponse = hearingEntity
-            .getHearingResponses().stream().filter(hearingResponseEntity ->
-                                                       hearingResponseEntity.getResponseVersion().equals(version))
```

Comment:
Moved this into `HearingEntity` for re-use elsewhere, but think it's meant to be comparing against the **request** version provided in the response instead so made that change in the moved method.